### PR TITLE
Adds Skrell Shared Dreaming

### PR DESCRIPTION
**Pull request by TehFlaminTaco**
Implements skrell shared dreaming.

Whilst Unconscious and alive, skrell will enter the Srom, aka Dream, in which they telepathically link to one another.
In game, they will appear in a custom area, able to communicate with one another.
Whilst in Srom, skrell find it very hard to keep their own secrets, if not impossible, and will often 'mumble' important information.

Skrell Meta-clique HQ

### CODE REVIEW

File: code/modules/shareddream/brainghost.dm [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	name = "Brain Ghost"
+	desc = "A Telepathic connection."
+
+	alpha = 127
+
+	var/image/ghostimage = null
+	var/mob/living/carbon/human/body = null
+
+
+	New(var/mob/living/carbon/human/form)
```
Comment (Alberyk) [inline code/modules/shareddream/brainghost.dm]: Use Initialize() here instead.

File: code/modules/shareddream/brainghost.dm [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			return
+		..()
+		overlays += image(form.icon,form,form.icon_state)
+		overlays += form.overlays
+		name = form.real_name
+		loc = pick(dream_entries)
+		body = form
+
+
+	verb/awaken()
```
Comment (Alberyk) [inline code/modules/shareddream/brainghost.dm]: Don't use relative pathing.

File: code/modules/shareddream/brainghost.dm [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+		set category = "IC"
+
+		if(body.willfully_sleeping)
+			body.sleeping = max(body.sleeping - 5, 0)
+			body.willfully_sleeping = 0
+			src << "<span class='notice'>You release your concentration on sleep, allowing yourself to awake.</span>"
+		else
+			src << "<span class='warning'>You've already released concentration. Wait to wake up naturally.</span>"
+
+	Life()
```
Comment (Alberyk) [inline code/modules/shareddream/brainghost.dm]: Don't use relative pathing.

**Comment by ben10083** [comment]
Two things @TehFlaminTaco 
1. Please explain what you mean by "mumble important information"
2. Please put this in projects in the forum for feedback.

**Comment by SleepyWolfe** [comment]
> 2. Please put this in projects in the forum for feedback.

am I allowed to veto this? If I do, I veto this, it was requested by myself - the skrell loredev. It's based on lore writing I've done.

File: code/modules/shareddream/brainghost.dm [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	name = "Brain Ghost"
+	desc = "A Telepathic connection."
+
+	alpha = 127
+
+	var/image/ghostimage = null
+	var/mob/living/carbon/human/body = null
+
+
+/mob/living/brain_ghost/Initialize()
```
[the remark this answers is not in the document]
Comment (TehFlaminTaco) [reply code/modules/shareddream/brainghost.dm]: @Alberyk There's one under the check if this is placed in the body, should I move it to before that?

[the remark this answers is not in the document]
Comment (Alberyk) [reply code/modules/shareddream/brainghost.dm]: Yes.

[the remark this answers is not in the document]
Comment (Alberyk) [reply code/modules/shareddream/brainghost.dm]: Like this: 
```suggestion
/mob/living/brain_ghost/Initialize()
	 . = ..()
```

**Comment by TehFlaminTaco** [comment]
[Suggestion created](https://forums.aurorastation.org/topic/11103-implement-skrell-shared-dreaming/) as per request. I don't suspect there'll be much pushback however.

**Pull request merged**